src/orion/core/worker/algo_wrappers/algo_wrapper.py

```diff
@@ -0,0 +1,321 @@
+""" ABC for a wrapper around an Algorithm. """
+from __future__ import annotations
+
+import copy
+from contextlib import contextmanager
+from logging import getLogger as get_logger
+from typing import Any, Generic, Sequence, TypeVar
+
+from typing_extensions import ParamSpec
+
+from orion.algo.base import BaseAlgorithm
+from orion.algo.registry import RegistryMapping
+from orion.algo.space import Space
+from orion.core.worker.trial import Trial
+
+logger = get_logger(__name__)
+
+AlgoType = TypeVar("AlgoType", bound=BaseAlgorithm)
+P = ParamSpec("P")
+
+
+class AlgoWrapper(BaseAlgorithm, Generic[AlgoType]):
+    """Base class for a Wrapper around an algorithm.
+
+    Can apply some transformation to the incoming trials before they get passed to the underlying
+    algorithm, and apply the reverse transformation to the trials suggested by the algorithm
+    before they are suggested by the wrapper.
+    """
+
+    def __init__(self, space: Space, algorithm: AlgoType):
+        # NOTE: This field is created automatically in the BaseAlgorithm class.
+        super().__init__(space)
+        self._algorithm = algorithm
+        assert not isinstance(algorithm, dict), algorithm
+        self.registry_mapping = RegistryMapping(
+            original_registry=self.registry,
+            transformed_registry=self.algorithm.registry,
+        )
+
+    def transform(self, trial: Trial) -> Trial:
```

Comment:
Do you think trial transformations are general enough to be part of the base algo wrapper? I'm not sure, but I don't have a strong opinion.

Reply:
I mean, we could have a `TransformWrapper` as an an intermediate base class, so that we'd pool the `SpaceAdapterWrapper --> TransformWrapper --> AlgoWrapper`, but that seems a bit unnecessary to me atm, since we only have a few wrappers.

Reply:
Yeah that's fine. I didn't realize at that point in my review that the wrapper for warm-starting would make transformations as well.